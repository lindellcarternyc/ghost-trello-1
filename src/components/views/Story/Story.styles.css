.story {
  border: 2px solid black; }
  .story .story--image--wrapper {
    border-bottom: 2px solid black; }
  .story .story--text {
    padding: 0.25rem 1rem; }
    .story .story--text .story--text--section {
      margin-bottom: 3rem; }
      .story .story--text .story--text--section p {
        margin: none; }
