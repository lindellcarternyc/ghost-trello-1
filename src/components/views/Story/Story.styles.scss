.story {
  border: 2px solid black;

  .description {
    padding: 0.25rem 1rem;
  }
}