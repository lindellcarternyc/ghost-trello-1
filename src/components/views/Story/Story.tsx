import * as React from 'react'

import './Story.styles.css'

interface StoryProp {
  title: string 
  image: string
  text: string
}

interface StoryProps {
  story: StoryProp
}

export const Story = (props: StoryProps) => {
  const { story } = props

  const { title, image, text } = story

  const sections = text.split('\n\n\n').map((section, idx) => {
    const paragraphs = section.split('\n \r').map((para, idy) => {
      return (
        <p key={idy}>{para}</p>
      )
    })
    return (
      <div key={idx} className='story--text--section'>
        {paragraphs}
      </div>
    )
  })
  return (
    <div className='story'>
      <div className='story--image--wrapper'>
        <img src={image} className='story--image' />
      </div>
      <h2>{title}</h2>
      <div className='story--text'>
        {sections}
      </div>
    </div>
  )
}