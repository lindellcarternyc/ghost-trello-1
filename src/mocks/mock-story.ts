import * as faker from 'faker'

export const generateTextSection = (): string => {
  const numParagraphGroups = faker.random.number({min: 3, max: 5})
  
  let section: string[] = []
  for (let i = 0; i < numParagraphGroups; i++) {
    const numParagraphsInGroup = faker.random.number({min: 4, max: 10})
    const paragraphGroup = faker.lorem.paragraphs(numParagraphsInGroup)
    section.push(paragraphGroup)
  }

  return section.join('\n\n\n')
}

export const generateHeaderImage = (): string => {
  return faker.image.abstract()
}

export const generateTitle = (): string => {
  let p1 = faker.lorem.sentence(3, 5)
  p1 = p1.slice(0, p1.length - 1) + ': '

  let p2 = faker.lorem.sentence(6, 10)
  p2 = p2.slice(0, p2.length - 1)
  return p1 + p2
  // return ''
}

export const generateAuthor = (): string => {
  return faker.name.findName()
}

const MONTHS = [
  'January', 'February', 'March', 'April', 'May',
  'June', 'July', 'August', 'September', 'October',
  'November', 'December'
]

export const generateWrittenDate = () => {
  const dateObj = faker.date.past(1)
  const year = dateObj.getFullYear()
  const month = MONTHS[dateObj.getMonth()]
  const date = dateObj.getDate()
  return `${month} ${date}, ${year}`
}

export const generateStory = (): {
  title: string, author: string, written: string, image: string, text: string
} => {
  const title = generateTitle()
  const author = generateAuthor()
  const written = generateWrittenDate()
  const image = generateHeaderImage()
  const text = generateTextSection()

  return {
    title,
    author,
    written,
    image,
    text
  }

}