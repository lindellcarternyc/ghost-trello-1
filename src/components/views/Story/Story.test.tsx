import * as React from 'react'
import * as ReactDOM from 'react-dom'
import * as renderer from 'react-test-renderer'

import { Story } from './Story'

import { generateDescription } from '../../../mocks/mock-story'

const MOCK_STORY = {
  description: generateDescription()
}

// describe('Story', () => {
it('renders a story without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Story description={MOCK_STORY.description}/>, div)
})

it('accepts a description prop that\'s a string', () => {
  const story = renderer.create(
    <Story description={MOCK_STORY.description} />
  )
  const tree = story.toJSON()
  expect(tree).toMatchSnapshot()
})
// })