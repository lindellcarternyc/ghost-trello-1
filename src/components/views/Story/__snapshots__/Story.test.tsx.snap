// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts a description prop that's a string 1`] = `
<div
  className="story"
>
  <p
    className="description"
  >
    hello
  </p>
</div>
`;
